webrender/src/picture.rs

```diff
@@ -18,6 +19,19 @@ use resource_cache::CacheItem;
 use scene::{FilterOpHelpers, SceneProperties};
 use tiling::RenderTargetKind;
 
+
+// TODO(gw): Rounding the content rect here to device pixels is not
+// technically correct. Ideally we should ceil() here, and ensure that
+// the extra part pixel in the case of fractional sizes is correctly
+// handled. For now, just use rounding which passes the existing
+// Gecko tests.
```

Comment:
The comment should be updated to indicate that width/height of 0 causes badness, hence the max'ing with 1. But other than that, LGTM